open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/manage/vrp/VRPConsentRequestValidatorTest.java

```diff
@@ -0,0 +1,2317 @@
+import com.wso2.openbanking.accelerator.consent.extensions.common.ConsentExtensionConstants;
+import com.wso2.openbanking.accelerator.consent.extensions.manage.model.ConsentManageData;
+import com.wso2.openbanking.accelerator.consent.extensions.manage.validator.VRPConsentRequestValidator;
+import com.wso2.openbanking.accelerator.consent.extensions.util.ConsentManageUtil;
+import com.wso2.openbanking.accelerator.consent.extensions.utils.ConsentExtensionTestUtils;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.JSONValue;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.wso2.openbanking.accelerator.consent.extensions.manage.validator.VRPConsentRequestValidator.validateAmountCurrency;
+import static com.wso2.openbanking.accelerator.consent.extensions.manage.validator.VRPConsentRequestValidator.validateAmountCurrencyPeriodicLimits;
+import static org.mockito.Mockito.mock;
+import static org.testng.AssertJUnit.assertFalse;
+
+/**
+ * test.
+ */
+@PowerMockIgnore({"jdk.internal.reflect.*"})
+@PrepareForTest({OpenBankingConfigParser.class})
+public class VRPConsentRequestValidatorTest extends PowerMockTestCase {
+
+    @Mock
+    private ConsentManageData consentManageData;
+
+    @Mock
+    OpenBankingConfigParser openBankingConfigParser;
+
+    private static Map<String, String> configMap;
+
+    @BeforeClass
+    public void setUp() throws ReflectiveOperationException {
+        MockitoAnnotations.initMocks(this);
+
+        configMap = new HashMap<>();
+        //to execute util class initialization
+        new CarbonUtils();
+        System.setProperty("some.property", "property.value");
+        System.setProperty("carbon.home", ".");
+        ConsentExtensionTestUtils.injectEnvironmentVariable("CARBON_HOME", ".");
+
+        consentManageData = mock(ConsentManageData.class);
+    }
+
+    @BeforeMethod
+    public void initMethod() {
+
+        openBankingConfigParser = mock(OpenBankingConfigParser.class);
+        Mockito.doReturn(configMap).when(openBankingConfigParser).getConfiguration();
+
+        PowerMockito.mockStatic(OpenBankingConfigParser.class);
+        PowerMockito.when(OpenBankingConfigParser.getInstance()).thenReturn(openBankingConfigParser);
+
+    }
+
+    @DataProvider(name = "vrpDataProvider")
+    public Object[][] vrpMetadataData() {
+        return VRPDataProviders.DataProviders.METADATA_DATA_HOLDER;
```

Comment:
Do we need these data providers? We use data providers to use same test method to test for different formats